apps/agenda/agenda.js

```diff
@@ -69,16 +75,18 @@ function showEvent(ev) {
   //var lines = [];
   if (ev.title) lines = g.wrapString(ev.title, g.getWidth()-10);
   var titleCnt = lines.length;
-  var start = getDate(ev.timestamp);
-  var end = getDate((+ev.timestamp) + (+ev.durationInSeconds));
+  var start = getDate(ev.timestamp, ev.allDay);
+  // All day events end at the midnight boundary of the following day. Here, we
+  // subtract one second for all day events so the days display correctly.
+  var end = getDate((+ev.timestamp) + (+ev.durationInSeconds) - ev.allDay, ev.allDay);
   var includeDay = true;
   if (titleCnt) lines.push(""); // add blank line after title
   if(start.getDay() == end.getDay() && start.getMonth() == end.getMonth())
     includeDay = false;
-  if(includeDay && ev.allDay) {
-    //single day all day (average to avoid getting previous day)
+  if(!includeDay && ev.allDay) {
+    //single day all day
```

Comment:
Good catch, that was actually not too readable and I got myself confused.. I wonder if we should change it to something like `sameDay`, that'd probably be more meaningful

Reply:
I agree that this should be renamed, but `sameDay` would reverse the current boolean's values, and that variable is used in quite a few places that wouldhave to be fixed.

I was already planning on doing a second pass with some cleanup like this to help make the code a little more clear, so I'll take care of the variable renaming then to avoid having too many changes in this PR, which might make it harder to understand.

Reply:
Yes good idea, feel free to add me as a reviewer